doc/src/manual/functions.md

```diff
@@ -158,16 +158,18 @@ Under the name `f`, the function does not support infix notation, however.
 
 A few special expressions correspond to calls to functions with non-obvious names. These are:
 
-| Expression        | Calls                  |
-|:----------------- |:---------------------- |
-| `[A B C ...]`     | [`hcat`](@ref)       |
-| `[A; B; C; ...]`  | [`vcat`](@ref)       |
-| `[A B; C D; ...]` | [`hvcat`](@ref)      |
-| `A'`              | [`adjoint`](@ref) |
-| `A.'`             | [`transpose`](@ref)  |
-| `1:n`             | [`colon`](@ref)      |
-| `A[i]`            | [`getindex`](@ref)   |
-| `A[i]=x`          | [`setindex!`](@ref)  |
+| Expression        | Calls                   |
+|:----------------- |:----------------------- |
+| `[A B C ...]`     | [`hcat`](@ref)          |
+| `[A; B; C; ...]`  | [`vcat`](@ref)          |
+| `[A B; C D; ...]` | [`hvcat`](@ref)         |
+| `A'`              | [`adjoint`](@ref)       |
+| `A.'`             | [`transpose`](@ref)     |
+| `1:n`             | [`colon`](@ref)         |
+| `A[i]`            | [`getindex`](@ref)      |
+| `A[i] = x`        | [`setindex!`](@ref)     |
+| `A.n`             | [`getproperty`](@ref Base.getproperty) |
+| `A.n = x`         | [`setproperty!`](@ref Base.setproperty!) |
```

Comment:
We really need more documentation on how to use this, explaining the arguments to `getproperty`, how to fall back to `getfield`, implications for performance and type stability, best practices....

Reply:
Also, there are zero tests, so some tests would be good...

Reply:
@stevengj FWIW I played with this a bit last night for PyCall, and just put up what I have so far in case it helps (won't be able to look again until tonight or tomorrow). The `getfield` fallback is clearly suboptimal -- we should probably change all of the `PyObject.o` -- and no `setproperty!` support, but the following works:

```
julia> np = pyimport("numpy")
PyObject <module 'numpy' from '/Users/inorton/.julia/v0.7/Conda/deps/usr/lib/python2.7/site-packages/numpy/__init__.pyc'>

julia> np.sin(1:100);
```

https://github.com/ihnorton/PyCall.jl/tree/getfield_parasido

Reply:
@ihnorton, thanks for looking at this!  Yes, my plan is to release a PyCall 2.0 for Julia 0.7, and switch over completely to `getproperty`.  But I'm not planning to work on it until 0.7 is close to release.

Reply:
> Also, there are zero tests, so some tests would be good...

I wrote them. Apparently I forgot to forgot to push them.